app/vue/src/client/preview/util.ts

```diff
@@ -1,19 +1,19 @@
-function getType(fn) {
+function getType(fn: Function) {
   const match = fn && fn.toString().match(/^\s*function (\w+)/);
   return match ? match[1] : '';
 }
 
 // https://github.com/vuejs/vue/blob/dev/src/core/util/props.js#L92
-function resolveDefault({ type, default: def }) {
+function resolveDefault({ type, default: def }: any) {
   if (typeof def === 'function' && getType(type) !== 'Function') {
-    // known limitation: we dont have the component instance to pass
+    // known limitation: we don't have the component instance to pass
     return def.call();
   }
 
   return def;
 }
 
-export function extractProps(component) {
+export function extractProps(component: any) {
```

Comment:
potentially can use `VueComponent` type from Vue types?

Reply:
This is where I could use your help @leoyli, would you like to give that a try and add to this PR?

Reply:
In storybook we're doing something custom with this `.options` I think it's not vue behavior. I don't think the value actually is a Vue.Component here..

Reply:
oh I see... so it's a bit hard to tell what really is based on that snippets of code